src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/HaplotypeCaller.java

```diff
@@ -160,14 +161,18 @@ public List<ReadFilter> getDefaultReadFilters() {
     }
 
     /**
-     * This is being used to set the mapping quality filter when in dragen mode... there are problems here...
+     * This is being used to set the mapping quality filter when in dragen mode. This is also where make alterations to the input arguments based on DragenMode.
      */
     protected String[] customCommandLineValidation() {
         if (hcArgs.dragenMode) {
             final GATKReadFilterPluginDescriptor readFilterPlugin =
                     getCommandLineParser().getPluginDescriptor(GATKReadFilterPluginDescriptor.class);
             Optional<ReadFilter> filterOptional = readFilterPlugin.getResolvedInstances().stream().filter(rf -> rf instanceof MappingQualityReadFilter).findFirst();
             filterOptional.ifPresent(readFilter -> ((MappingQualityReadFilter) readFilter).minMappingQualityScore = 1);
+            ModeArgumentUtils.setArgValues(
+                    getCommandLineParser(),
+                    hcArgs.getDragenNameValuePairs(),
+                    HaplotypeCallerArgumentCollection.DRAGEN_GATK_MODE_LONG_NAME);
```

Comment:
Would it be a good idea for `setArgValues()` itself to check whether the provided mode is activated, and proceed with setting the args only if it's active? Or is an external check on the mode like you have here preferable? Either way, we don't need to address in this PR.